lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -66,6 +66,34 @@ class AwsCompileFunctions {
     }
   }
 
+  downloadS3Artifact(functionName) {
+    const functionObject = this.serverless.service.getFunction(functionName);
+    const artifactFilePath = _.get(functionObject, 'package.artifact') ||
+      _.get(this, 'serverless.service.package.artifact');
+
+    const regex = new RegExp('.*s3.amazonaws.com/(.+)/(.+)');
+    const match = artifactFilePath.match(regex);
+
+    if (match) {
+      return this.provider.request('S3', 'getObject', {
+        Bucket: match[1],
+        Key: match[2],
+      }).then((res) => {
```

Comment:
This approach seems not very efficient. As we load entire zip (which can be large), into process memory  and then flush it down to FS.

It'll be more performant and nicer for a process memory, to just stream it down directly to FS

Reply:
Actually I tried it but in that case we have to use the AWS SDK directly rather than going through `this.provider.request` since it'll return a Buffer if you don't call their streaming function on their request object.

I'll update the code today to use their streaming functionality.